### ISSUE TITLE
Set intersect query with []string and .In()

### ISSUE THREAD

**Issue by Southclaws** [prisma/prisma-client-go#594]
I'm working with a model that includes a `tags []string` kind of thing, and I'd like to know if it's possible to do some specific intersections for queries.

For reference, the feature I'm working on is open source so it may help to check the PR: https://github.com/openmultiplayer/web/pull/420/files

The relevant part of the schema is here: https://github.com/openmultiplayer/web/blob/forum-categories/prisma/schema.prisma#L94-L128 notably the list of `Tag`s in each `Post` model.

It works great with these queries: https://github.com/openmultiplayer/web/pull/420/files#diff-3c9c79132aaec9a342a9567721f83d4d8b968069440955ebdbf80e89bd31b1b1R155-R183 if you specify one or many tags in the arguments, you get **any** post that has **any** of the specified tags.

So, let's say I have 3 posts
1. tagged "Food"
2. tagged "Music"
3. tagged "Food" and "Music"

When I query "food,music" I get post 1 and 2 but not post 3

When I query "food" I get post 1 but not post 3

When I query "music" I get post 2 but not post 3

So it seems posts with multiple tags cannot be queried with my approach.

Is there a good way of doing this with the current Go API? Or do I have to go to raw SQL for this.

Thanks!

**Comment by steebchen** [comment]
I can confirm this, something is wrong internally here as the filters for in should actually not have `In` and array-specific methods (e.g. `HasSome`). I will try to pick this up and fix it soon.

In the meanwhile, I think you could try using an `.Or()` and then `.Equals(tag)` for each tag. Something like this:

```go
queries := []db.PostWhereParam{
	db.Post.First.Equals(true),
	db.Post.CreatedAt.Before(before),
}
var tagQueries []db.PostWhereParam
for _, tag := range tags {
	tagQueries = append(tagQueries, db.Post.Tags.Some(db.Tag.Name.Equals(tag)))
}
queries = append(
	queries,
	db.Post.Or(
		tagQueries...
	),
)
// ...FindMany(queries)...
```

I know, this is ugly, but it'll save you doing two queries though :P

**Comment by Southclaws** [comment]
Hey, thanks for the response! This looks like a good alternative, it makes sense, but I just tried it and it doesn't seem to work.

I even tried hard-coding the query, just to be sure:

```go
				FindMany(
					db.Post.First.Equals(true),
					db.Post.CreatedAt.Before(before),
					db.Post.Or(
						db.Post.Tags.Some(
							db.Tag.Name.Contains("open.mp"),
						),
					),
				).
```

This returns two items, which _only_ have the tag `open.mp` but it doesn't return the item that has _two_ tags, `open.mp` and `forum`:

![image](https://user-images.githubusercontent.com/1636971/135661129-603d7617-b888-49fe-b012-ddb62e9fba19.png)

![image](https://user-images.githubusercontent.com/1636971/135661233-78f20e2c-83f4-40b2-8cac-7ef849f11cbe.png)

**Comment by Southclaws** [comment]
Actually, I'm just going to remove the other filters (before date and first post) to make sure those aren't filtering it out accidentally.

**Comment by Southclaws** [comment]
Awesome thanks! I will give this a try when I get time and let you know how it goes 👍

**Comment by steebchen** [comment]
Are you sure you're using the main branch version? There is no release yet.

**Comment by steebchen** [comment]
Ah. That's a relation field and not an array/list field. You need to a relation query with .Where and then do .HasSome on the field of type string[]. Basically the same query as before except of doing .In you write .HasSome

**Comment by steebchen** [comment]
Hm did your schema change? I'll check again tomorrow when I'm back on my computer

**Comment by Southclaws** [comment]
Ohh because my tags list is a relation not a simple list of strings? I completely forgot this detail while writing the issue!

**Comment by steebchen** [comment]
Yeah exactly. These operations only exist for list/array operations. In your case it seems you just have a another model (`Tag`). You still should be able to query for what you want either way, e.g. with .Some and then compare the Tag.name with .Equals or something.

**Comment by steebchen** [comment]
You might want to think about if maybe a string arrayed  would be sufficient, but that obviously depends on your use-case. I think it would fit considering your current schema but would not work anymore if e.g. users could have their own tags on posts.

**Comment by Southclaws** [comment]
Yeah the reason for this is to show all posts for a particular tag etc. also we might add colours to tags too or other metadata so it won't fit a []string.

I'll try what I originally tried which was `db.Post.Tags.Some(db.Tag.Name.In(tags)),` or the earlier suggestion you posted here https://github.com/prisma/prisma-client-go/issues/594#issuecomment-927948361
